lmcache/address_manager/disk_address_manager.py

```diff
@@ -0,0 +1,285 @@
+        disk_path = config.get("disk_path", "/local_disk/")
+
+        return LMCAddressManagerConfig(disk_url, disk_path)
+
+
+class LMCDiskAddressManager():
+    """
+    Cache engine for storing the KV cache of the tokens in the local disk.
+    """
+
+    def __init__(self, config: LMCAddressManagerConfig):
+        """
+        Throws:
+            RuntimeError if the loaded configuration does not match the current
+                configuration
+        """
+
+        # Dict key to path & size
+        self.dict: OrderedDict[CacheEngineKey,
+                               DiskCacheMetadata] = OrderedDict()
+
+        self.path = config.disk_path
+
+        assert self.path is not None, ("Need to specify local path if when "
+                                       "using  Local Disk")
+
+        if not os.path.exists(self.path):
+            os.makedirs(self.path)
+
+        self.update_lock = threading.Lock()
+
+        self.evictor = LRUEvictor()
+
+    def _key_to_path(
+        self,
+        key: CacheEngineKey,
+    ) -> str:
+        """
+        Convert key to path_name
+
+        Input:
+            key: the key of the token chunk, including prefix hash and format
+
+        Returns:
+            returns the path name
+        """
+        return self.path + key.to_string().replace("/", "-") + ".pt"
+
+    def remove(
+        self,
+        key: CacheEngineKey,
+    ) -> None:
+        """
+        Remove the KV cache chunk by the given key
+
+        Input:
+            key: the key of the token chunk, including prefix hash and format
+
+        """
+
+        self.update_lock.acquire()
+        path = self.dict[key].path
+        self.dict.pop(key)
+        self.update_lock.release()
+
+        os.remove(path)
+
+    def contains(
+        self,
+        key_str: str,
+    ) -> str:
+        """
+        Check if the cache engine contains the key.
+
+        Input:
+            key: the key of the token chunk, including prefix hash and format
+
+        Returns:
+            path if the cache engine contains the key, "" otherwise
+        """
+        key = CacheEngineKey.from_string(key_str)
+        with self.update_lock:
+            return self.dict[key].path if key in self.dict else ""
+
+    def batched_contains(
+        self,
+        keys_strs: Iterable[str],
+    ) -> Iterable[str]:
+        return [self.contains(key_str) for key_str in keys_strs]
+
+    def write_check(self, key_str: str, kv_size: float) -> str:
+        """
+        The logic here is that:
+            return "" if the key is already in the cache, 
+                so that backend will not write it again
+            return path if the key is not in the cache, 
+                so that backend will write it to the path
+        if path is not in the cache, initialize the cache with ("",0)
+        """
+        key = CacheEngineKey.from_string(key_str)
+        with self.update_lock:
+            if key in self.dict:
```

Comment:
IIUC, the code here has a bug:
process A calling `write_check` with `key_str` --> `self.dict[key]` is assigned with `DiskCacheMetadata("", 0)`
process B calling `write_check` or `read_check` with  the same `key_str` --> return the path.

However, it's possible that process A hasn't written any thing into the disk yet, and race condition will happen (process B will try read / write a non-exist file)

Reply:
I didn't get it.

Reply:
Nop. This will not happens. process B calling write_check or read_check with the same key_str will get the path "", which also means not exist